sky/clouds/aws.py

```diff
@@ -414,9 +447,22 @@ def get_current_user_identity(self) -> Optional[str]:
         return user_id
 
     def get_credential_file_mounts(self) -> Dict[str, str]:
+        # ~/.aws/credentials is required for users using multiple clouds.
+        # If this file does not exist, users can launch on AWS via AWS SSO and assign
+        # IAM role to the cluster.
+        # However, if users launch clusters in a non-AWS cloud, those clusters do not
+        # understand AWS IAM role so will not be able to access private AWS EC2 resources
+        # and S3 buckets.
+
+        # The file should not be uploaded if the user is using SSO, as
+        # the credential file can be from a different account, and will
+        # make autopstop/autodown/spot controller misbehave.
+        if self._is_sso_identity():
```

Comment:
Q: For a non-SSO user, can / should we also avoid uploading the credential files to her new AWS clusters, and rely on the V2 node provider which assigns the new IAM role?

Reply:
Ah, this code path is called regardless of what cloud the new cluster is being launched on. That's the reason we don't do the above, right?

I think it may be fine to add a NOTE/TODO here saying we can have a mechanism to know what cloud the new cluster is being provisioned in, and skip uploading this for new AWS clusters.

Reply:
Good point! Added a TODO for future refactoring.